web/src/typing.js

```diff
@@ -79,11 +111,19 @@ function get_current_time() {
 }
 
 function notify_server_start(to) {
-    send_typing_notification_based_on_message_type(to, "start");
+    if (typeof to === "number") {
+        send_typing_notifications_for_message_edit(to, "start");
+    } else {
+        send_typing_notification_based_on_message_type(to, "start");
+    }
```

Comment:
We're not going to merge a version which uses these hackery `number` types to determine what type of object this is. Let's get this file migrated to `TypeScript` and use actual types for our local data structures.